packages/design-system/src/components/combobox.tsx

```diff
@@ -77,32 +118,26 @@ type useComboboxProps<Item> = {
   ) => Partial<UseComboboxStateChangeOptions<Item>>;
 };
 
-// eslint-disable-next-line func-style
-export function useCombobox<Item>({
+export const useCombobox = <Item,>({
```

Comment:
I think it's a stupid rule to have. What's wrong with function keyword? How is the const better considering the ugly TS syntax?

Reply:
Another option is to enable Item extends unknown in the linter, so that it looks it least less like a hack. If you really want to discuss function keyword, we can do that separately

Reply:
I actually don't want to discuss code style since it has 0 or even negative value. Spending time on fixing such issues just because we have this rule is a waste of everyone's time IMO. But it looks like you have a strong option about it so let's keep doing what we doing 🤷🏻‍♂️

Reply:
I agree, it doesn't really matter, I feel strongly about it because I am trying to enforce a more consistent structure of code, meaning the order of definition - first define, then use. With named functions people can create very different structure and make it harder to read overall, you can't scan the code consistently.
When a few people work on the project - it doesn't matter. At scale this flexibility makes it harder to read.
In this particular case it wouldn't matter, because its a small file that will stay small etc. But someone will look it up and do the same somewhere else, where its not a good thing any more. So I am trying to set an example.